pkg/cloudevents/transport/http/message.go

```diff
@@ -77,3 +81,32 @@ func (m Message) CloudEventsVersion() string {
 
 	return ""
 }
+
+func readAllClose(r io.ReadCloser) ([]byte, error) {
+	if r != nil {
+		defer r.Close()
+		return ioutil.ReadAll(r)
+	}
+	return nil, nil
+}
+
+// FromRequest initializes a Message from a http.Request
+func (m *Message) FromRequest(r *http.Request) (err error) {
```

Comment:
I think this method signature should be:

```golang
func FromRequest(req *http.Request)  (*Message, error)
```

Reply:
I prefer to initialize in place because it gives the user more control over allocations. In a high-performance event bridge, it means the difference between a pool of re-usable Message objects and an allocation for every message, right on the critical path.  I've added a NewMessage() constructor with the signature you requested, but I'd rather keep in-place initialization.